xds/internal/balancer/ringhash/e2e/ringhash_balancer_test.go

```diff
@@ -134,11 +155,797 @@ func (s) TestRingHash_ReconnectToMoveOutOfTransientFailure(t *testing.T) {
+	hashPolicy := v3routepb.RouteAction_HashPolicy{
+		PolicySpecifier: &v3routepb.RouteAction_HashPolicy_FilterState_{
+			FilterState: &v3routepb.RouteAction_HashPolicy_FilterState{
+				Key: "io.grpc.channel_id",
+			},
+		},
+	}
+	action := route.VirtualHosts[0].Routes[0].Action.(*v3routepb.Route_Route)
+	action.Route.HashPolicy = []*v3routepb.RouteAction_HashPolicy{&hashPolicy}
+	return route
+}
+
+// checkRPCSendOK sends num RPCs to the client. It returns a map of backend
+// addresses as keys and number of RPCs sent to this address as value. Abort the
+// test if any RPC fails.
+func checkRPCSendOK(t *testing.T, ctx context.Context, client testpb.TestServiceClient, num int) map[string]int {
+	t.Helper()
+
+	backendCount := make(map[string]int)
+	for i := 0; i < num; i++ {
+		var remote peer.Peer
+		_, err := client.EmptyCall(ctx, &testpb.Empty{}, grpc.Peer(&remote))
+		if err != nil {
+			t.Fatalf("rpc UnaryCall() failed: %v", err)
+		}
+		backendCount[remote.Addr.String()]++
+	}
+	return backendCount
+}
+
+// TestRingHash_AggregateClusterFallBackFromRingHashAtStartup tests that when
+// an aggregate cluster is configured with ring hash policy, and the first
+// cluster is in transient failure, all RPCs are sent to the second cluster
+// using the ring hash policy.
+func (s) TestRingHash_AggregateClusterFallBackFromRingHashAtStartup(t *testing.T) {
+	xdsServer, nodeID, _, xdsResolver, stop := e2e.SetupManagementServer(t, e2e.ManagementServerOptions{
+		AllowResourceSubset: true,
+	})
+	defer stop()
+
+	nonExistantServers, stopNonExistant := startTestServiceBackends(t, 2)
+	servers, stop := startTestServiceBackends(t, 2)
+	defer stop()
+
+	// Stop the servers that we want to be unreachable.
+	stopNonExistant()
+
+	newCluster1Name := "new_cluster_1"
+	newEdsService1Name := "new_eds_service_1"
+	newCluster2Name := "new_cluster_2"
+	newEdsService2Name := "new_eds_service_2"
+	clusterName := "aggregate_cluster"
+
+	ep1 := e2e.EndpointResourceWithOptions(e2e.EndpointOptions{
+		ClusterName: newEdsService1Name,
+		Localities: []e2e.LocalityOptions{{
+			Name:     "locality0",
+			Weight:   1,
+			Backends: backendOptions(t, nonExistantServers),
+		}},
+	})
+	ep2 := e2e.EndpointResourceWithOptions(e2e.EndpointOptions{
+		ClusterName: newEdsService2Name,
+		Localities: []e2e.LocalityOptions{{
+			Name:     "locality0",
+			Weight:   1,
+			Backends: backendOptions(t, servers),
+		}},
+	})
+	cluster1 := e2e.ClusterResourceWithOptions(e2e.ClusterOptions{
+		ClusterName: newCluster1Name,
+		ServiceName: newEdsService1Name,
+	})
+	cluster2 := e2e.ClusterResourceWithOptions(e2e.ClusterOptions{
+		ClusterName: newCluster2Name,
+		ServiceName: newEdsService2Name,
+	})
+	cluster := e2e.ClusterResourceWithOptions(e2e.ClusterOptions{
+		ClusterName: clusterName,
+		Type:        e2e.ClusterTypeAggregate,
+		// TODO: when "A75: xDS Aggregate Cluster Behavior Fixes" is merged, the
+		// policy will have to be set on the child clusters.
```

Comment:
Nit: Technically, not when A75 is merged, but when support for that is merged in the grpc-go repo.

Reply:
I replaced "merged" with "implemented".